sky/resources.py

```diff
@@ -1117,10 +1117,75 @@ def get_required_cloud_features(
         return features
 
     @classmethod
-    def from_yaml_config(cls, config: Optional[Dict[str, str]]) -> 'Resources':
+    def from_yaml_config(
+        cls, config: Optional[Dict[str, Any]]
+    ) -> Union[Set['Resources'], List['Resources']]:
         if config is None:
-            return Resources()
+            return {Resources()}
+
+        def _override_resources(base_resource_config,
+                                override_configs) -> List[Resources]:
+            resources_list = []
+            for override_config in override_configs:
+                new_resource_config = base_resource_config.copy()
+                new_resource_config.update(override_config)
+                resources_list.append(
+                    Resources._from_yaml_config_single(new_resource_config))
+            return resources_list
+
+        config = config.copy()
+        any_of_configs = config.pop('any_of', None)
+        ordered_configs = config.pop('ordered', None)
+        if any_of_configs and ordered_configs:
```

Comment:
nit: use is not None?

Reply:
I was not using `is not None`, as I wanted to allow empty field, but it seems not necessary. Added. Thanks!